### DISCUSSION TITLE
Is it possible to alter ForeignKey object?

### DISCUSSION THREAD

**Discussion by Gayathri-A-R** [sqlalchemy/alembic#1116] [Usage Questions]
Is there any way to alter target_fullname and column of ForeignKey objects or to specify a different schema for targeted table.

There are some cases due to which I cant specify **'schema.table.col'** in models.py. I can only specify **'table.col'** for which error is thrown saying targeted table not found. 

Use case: https://github.com/sqlalchemy/alembic/discussions/1105#discussioncomment-3932371

**Comment by zzzeek** [comment]
you can't modify `ForeignKey` directly but you can use a schema name that is replaceable, using the [Schema Translate](https://docs.sqlalchemy.org/en/14/core/connections.html#schema-translating) feature.  however, Alembic doesn't generally pick up on "schema translate" right now within `op` directives like `alter_column`.   It should work for a `op.create_table` action, which is where you'd see `ForeignKey` used explicitly, though i might opt for just writing my migration scripts to use a variable like `ForeignKey(f"{schemaname}.tablename")` and pass through `schemaname` from env.py to the migration script.
